src/main/java/pl/wrapper/parking/infrastructure/inMemory/InMemoryRepositoryImpl.java

```diff
@@ -0,0 +1,63 @@
+package pl.wrapper.parking.infrastructure.inMemory;
+
+import jakarta.annotation.PostConstruct;
+import jakarta.annotation.PreDestroy;
+
+import org.springframework.core.serializer.support.SerializationFailedException;
+import org.springframework.scheduling.annotation.Scheduled;
+
+import java.io.*;
+import java.util.Map;
+
+public abstract class InMemoryRepositoryImpl<K extends Serializable, V extends Serializable> implements InMemoryRepository<K, V> {
+    protected final transient File file;
+    protected Map<K, V> dataMap;
```

Comment:
let's change List to map. Using indexes is a bad idea - especially since they rely on the order of adding the data, which can change